core/database_util.go

```diff
@@ -138,6 +189,168 @@ func GetBody(db ethdb.Database, hash common.Hash) *types.Body {
 	return body
 }
 
+// WriteBlockAddTxIndexes writes atx-indexes for a given block.
+func WriteBlockAddTxIndexes(indexDb ethdb.Database, block *types.Block) error {
+	batch := indexDb.NewBatch()
+	if _, err := putBlockAddrTxsToBatch(batch, block); err != nil {
+		return err
+	}
+	return batch.Write()
+}
+
+// putBlockAddrTxsToBatch formats and puts keys for a given block to a db Batch.
+// Batch can be written afterward if no errors, ie. batch.Write()
+func putBlockAddrTxsToBatch(putBatch ethdb.Batch, block *types.Block) (txsCount int, err error) {
+	for _, tx := range block.Transactions() {
+		txsCount++
+
+		from, err := tx.From()
+		if err != nil {
+			return txsCount, err
+		}
+		// Note that len 8 because uint64 guaranteed <= 8 bytes.
+		bn := make([]byte, 8)
+		binary.LittleEndian.PutUint64(bn, block.NumberU64())
+
+		if err := putBatch.Put(formatAddrTxBytesIndex(from.Bytes(), bn, []byte("f"), tx.Hash().Bytes()), nil); err != nil {
+			return txsCount, err
+		}
+
+		to := tx.To()
+		if to == nil || to.IsEmpty() {
+			continue
+		}
+		if err := putBatch.Put(formatAddrTxBytesIndex(to.Bytes(), bn, []byte("t"), tx.Hash().Bytes()), nil); err != nil {
+			return txsCount, err
+		}
+	}
+	return txsCount, nil
+}
+
+// GetAddrTxs gets the indexed transactions for a given account address.
+func GetAddrTxs(db ethdb.Database, address common.Address, blockStartN uint64, blockEndN uint64, direction string) []string {
+	if direction != "to" && direction != "from" && direction != "both" && direction != "" {
```

Comment:
this could be refactored:

```go
if len(direction) > 0 && !strings.Contains("btf", direction[0]) {
```

which would allow passing t, f, b, to, from, both